src/Fixer/Strict/StrictTypesFixer.php

```diff
@@ -26,81 +27,142 @@
      */
     public function isCandidate(Tokens $tokens)
     {
-        return true;
+        return count($tokens) > 1 && $tokens[0]->isGivenKind(T_OPEN_TAG);
     }
 
     /**
      * {@inheritdoc}
      */
     public function fix(\SplFileInfo $file, Tokens $tokens)
     {
-        if (!$tokens->isMonolithicPhp()) {
+        // check for if declaration already in file
+        $sequence = $this->getDeclareStrictTypeSequence();
+        $searchIndex = $tokens->getNextMeaningfulToken(0);
+        if (null === $searchIndex) {
+            $this->insertNewSequence($tokens); // not found insert
+
             return;
         }
 
-        $declareTokens = array(
-            new Token(array(T_OPEN_TAG, '<?php ')),
-            new Token(array(T_DECLARE, 'declare')),
-            new Token('('),
-            new Token(array(T_STRING, 'strict_types')),
-            new Token('='),
-            new Token(array(T_LNUMBER, '1')),
-            new Token(')'),
-            new Token(';'),
-            new Token(array(T_WHITESPACE, "\n\n")),
-        );
-
-        // check for valid construct, break on mismatch
-        for ($i = 0;; ++$i) {
-            if (!$tokens[$i]->equals($declareTokens[$i], true)) {
-                break;
-            }
-            if ($i === count($declareTokens) - 1) {
-                return;
-            }
+        $sequenceLocation = $tokens->findSequence($sequence, $searchIndex, null, false);
+        if (null === $sequenceLocation) {
+            $this->insertNewSequence($tokens); // not found insert
```

Comment:
`// not found insert`
? Isn't something wrong with grammar here? what insert?

same later